File: 1400-Construct-K-Palindrome-Strings.py
class Solution:
    def canConstruct(self, s: str, k: int) -> bool:
        freq={}
        for char in s:
            if char in freq:
                freq[char]+=1
            else:
                freq[char]=1
        odd_count = 0
        for i in freq.values():
            if i%2==1:
                odd_count+=1

        return k <= len(s) and  k >= odd_count

File: 1400-Construct-K-Palindrome-Strings.cpp
class Solution {
public:
    bool canConstruct(string s, int k) {
        if (k>s.size()){return false;} 
        unordered_map<char, int> freq;
        for (char c: s){
            freq[c]++ ;
        }
        int count=0;
        for (auto i : freq) {
            if (i.second%2==1)
            count++ ;
        }
        if (k>=count) {return true;}
        else {return false;} 
    }
};